### PR TITLE
add characterEncoding parameter suppprt && add setBytes getBytes

### LINKED ISSUE #107: Can url be this "jdbc:clickhouse://127.0.0.1:9000?characterEncoding=UTF-8" ？i.e. Specify the Charset to encode

### PR DESCRIPTION
**Pull request by sundy-li**
closes #107  #169

### CODE REVIEW

File: clickhouse-native-jdbc/src/main/java/com/github/housepower/jdbc/settings/SettingKey.java [hunk 1]
```diff
@@ -146,7 +146,8 @@
     host(ISettingType.String, ""),
     database(ISettingType.String, ""),
     password(ISettingType.String, ""),
-    query_timeout(ISettingType.Seconds, "");
+    query_timeout(ISettingType.Seconds, ""),
+    characterEncoding(ISettingType.String, "charset for converting between Bytes and String");
```
Comment (pan3793) [inline clickhouse-native-jdbc/src/main/java/com/github/housepower/jdbc/settings/SettingKey.java:150]: `characterEncoding` is MySQL style parameter, we can use `charset` to unify parameter style

File: clickhouse-native-jdbc/src/test/java/com/github/housepower/jdbc/AbstractITest.java [hunk 2]
```diff
@@ -31,14 +32,19 @@
     protected static final int SERVER_PORT = Integer.parseInt(System.getProperty("CLICK_HOUSE_SERVER_PORT", "9000"));
 
     protected String getJdbcUrl() {
```
Comment (pan3793) [inline clickhouse-native-jdbc/src/test/java/com/github/housepower/jdbc/AbstractITest.java:34]: can we drop this method?

**Review by pan3793 — CHANGES_REQUESTED** [review]
Cool job, left few comments

**Review by pan3793 — APPROVED** [review]

**Comment by uschindler** [comment]
Thanks, looks great!

File: clickhouse-native-jdbc/src/main/java/com/github/housepower/jdbc/serializer/BinaryDeserializer.java [hunk 3]
```diff
@@ -66,6 +66,12 @@ public boolean readBoolean() throws IOException {
         return (container.get().readBinary() != 0);
     }
 
+    public byte[] readBytesBinary() throws IOException {
+        byte[] data = new byte[(int) readVarInt()];
+        container.get().readBinary(data);
+        return data;
+    }
+
     public String readStringBinary() throws IOException {
```
Comment (uschindler) [inline clickhouse-native-jdbc/src/main/java/com/github/housepower/jdbc/serializer/BinaryDeserializer.java:75]: is this method not obsolete now?

**Review by uschindler — COMMENTED** [review]
just a small comment

File: clickhouse-native-jdbc/src/main/java/com/github/housepower/jdbc/serializer/BinaryDeserializer.java (hunk 3 above)
Comment (uschindler) [reply clickhouse-native-jdbc/src/main/java/com/github/housepower/jdbc/serializer/BinaryDeserializer.java:75]: i am not really happy with this overload, as one returns a byte[] the other a string and you can't be sure which one is called.

Comment (sundy-li) [reply clickhouse-native-jdbc/src/main/java/com/github/housepower/jdbc/serializer/BinaryDeserializer.java:75]: No, I think we should use UTF-8 for the column name and the column type by default.
Do you agree with that?

```
String name = deserializer.readStringBinary();
```

Comment (uschindler) [reply clickhouse-native-jdbc/src/main/java/com/github/housepower/jdbc/serializer/BinaryDeserializer.java:75]: Ah it is for the column name. I agree with that, although most people will use US-ASCII only :-)

It maybe a good idea to rename the method returning the byte[].

Comment (uschindler) [reply clickhouse-native-jdbc/src/main/java/com/github/housepower/jdbc/serializer/BinaryDeserializer.java:75]: ah sorry, you renamed it. My fault!

Comment (sundy-li) [reply clickhouse-native-jdbc/src/main/java/com/github/housepower/jdbc/serializer/BinaryDeserializer.java:75]: >  i am not really happy with this overload, as one returns a byte[] the other a string and you can't be sure which one is called.

I just checked, ClickHouse is using `US-ASCII` for column name and type, so it's better to rename it to  `readUTF8StringBinary` as you suggested.

**Pull request merged**